Feat: Switch component (#135)
Atom-Learning/components: src/components/switch/Switch.mdx

---
title: Switch
component: Switch
description: An implementation of the Switch component from Radix with default styling and the css prop
category: Forms
---

> The Switch component implements [Radix's Switch component](https://radix-ui.com/primitives/docs/components/switch) where the default value can be set via `default` props.

Reviewer: I think the comment about the `default` prop feels like a bit of a non sequitur here. It's definitely worth covering in the docs (see other comments on this file) but I don't think it fits in this para.

```suggestion
The Switch component implements [Radix's Switch component](https://radix-ui.com/primitives/docs/components/switch).
```

> Switches are used to change the state of functionality and preferences. They prompt users to choose between two mutually exclusive options and will always have a default value.

Reviewer: Including these details about proper usage patterns is 🔥 but I think we can tighten it up a little to make it clearer:

```suggestion
Switches are used to toggle application state between two mutually exlusive values. They always have a default value.
```

> The Switch component should not be used to handle data (e.g. where we submit data in a form it's encouraged to use checkboxes or radio buttons). Instead it should be used interactively where the effect is an immediate display or state change.

Reviewer: This is all v good info but I think we can be a little clearer about what our users should take away from it:

```suggestion
The Switch component should not be used to collect form data — use checkboxes or radio buttons for that. Only use Switch inside a form if it is serving its primary purpose of toggling some state with immediate effect (e.g. to conditionally render a section of the form).
```

## Appearance

> ```tsx live

Reviewer 2: `live` isn't appropriate here, `preview` should be fine 👍 

I'd also maybe remove the `## Appearance` title to keep it consistent with the other docs

> <Switch />

Reviewer: The docs say that we need a default value but one isn't provided here. Can we clarify whether we should be providing one manually / what happens if we don't?

Author: Yeah that's my bad, that's an incorrect explanation and a hangover from the Toggle. I've changed the documentation to clarify
